src/txn/txn.c

```diff
@@ -1665,7 +1665,8 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
                     WT_ERR(__txn_resolve_prepared_op(session, op, true, &cursor));
 
                 /* Sleep between resolving prepared operations when configured. */
-                __wt_timing_stress(session, WT_TIMING_STRESS_PREPARE_RESOLUTION, NULL);
+                if (i < 10)
```

Comment:
Should we try to call this on random iterations instead of the first 10 ones?

Reply:
If we do it on random iterations there's a slim possibility of calling the timing stress too many times still.

Reply:
Yes, we would limit this as well making sure we don't stress too many times. My suggestion is to stress up to N times on possibly non-consecutive iterations.

Reply:
If we do that, should we add a check for the flag WT_TIMING_STRESS_PREPARE_RESOLUTION first to avoid doing a random number generation unnecessarily? If so should we check that flag here in this function or in a new wrapper function to `__wt_timing_stress`?

Reply:
I would do the check outside `__wt_timing_stress` as `__wt_timing_stress` is called in other places. I guess it could look like this:
```
if WT_TIMING_STRESS_PREPARE_RESOLUTION is set && num_times < threshold && random
    __wt_timing_stress(session, WT_TIMING_STRESS_PREPARE_RESOLUTION, NULL);
```

Reply:
There's some value in doing it randomly since it's then more likely to cover a variety of operation types, but doing it this way will still be biased towards the front of the operations list. If we're going to do it randomly I think it'd be preferable to say something like:

```
if (WT_TIMING_STRESS_PREPARE && (i % (txn->mod_count / 10)) == 0)
    __wt_timing_stress(etc);
```

Reply:
Does that work for a mod_count less than 10? Which it is most of the time?

Reply:
It's always true at `i == 0`, so you're always going to get at least one timing stress. I'm still just throwing ideas around, it's definitely be possible to improve on this - the idea was not to bias it towards one end of the operations list.

Reply:
Yeah, that makes sense

Reply:
I realise this won't work for any mod_count < 10, I'll continue to think about the best way to randomise this. Although I don't see any problem with the stress being biased towards the front. If the goal of the stress is to create a long-running prepare resolution then we care more about how long the entire loop is taking rather than each update.

Reply:
A mod_count < 10 won't work since we can't do `i % 0`. If we want to keep an even distribution of sleeps amongst the updates we could call sleep with some probability and add a counter `sleep_count` and be sure not to exceed some threshold of sleeps. I was trying to add much testing code to this function but what's more desirable here?

Reply:
Sounds good to me!